numpy/core/code_generators/generate_umath.py

```diff
@@ -904,7 +907,14 @@ def english_upper(s):
           "PyUFunc_SimpleBinaryOperationTypeResolver",
           TD(ints),
           TD('O', f='npy_ObjectLCM'),
-          )
+          ),
+'matmul' :
+    Ufunc(2, 1, None,
+          docstrings.get('numpy.core.umath.matmul'),
+          "PyUFunc_MatmulTypeResolver",
+          TD(notimes_or_obj),
+          signature='"(n?,k),(k,m?)->(n?,m?)"',
```

Comment:
Nit: These double quotes seem a little weird to me - I'd prefer to insert them as part of the code generation, not here.

Reply:
ok

Reply:
Requires an adjustment to the parsing code, rather than NULL the default signature is now `""`. I guess this is an improvement since now we will handle both.

Reply:
I wouldn't change the parsing code - I'm suggesting changing this line to
```
signature="(n?,k),(k,m?)->(n?,m?)"
```
And the code generator from
```
"... {sig}".format(sig=signature)
```
to
```
if signature is not None:
    sig = '"{}"'.format(signature)
else:
    sig = "NULL"
"... {sig}".format(sig=sig)
```

Reply:
ok